src/main/java/com/google/devtools/build/lib/syntax/BinaryOperatorExpression.java

```diff
@@ -189,6 +189,18 @@ private static Object evaluate(
         case PIPE:
           return pipe(x, y, env, location);
 
+        case AMPERSAND:
+          return (Integer) x & (Integer) y;
```

Comment:
If x is not an Integer, the program will crash. Each cast must be dominated by a successful instanceof check. Study the Go code for an example.

Please add a test that exercises this case.